images/macos/software-report/SoftwareReport.Common.psm1

```diff
@@ -15,101 +15,116 @@ function Get-GoVersion {
         $goOutput = $goOutput.Substring(2)
     }
 
-    return "Go $goOutput"
+    return "$goOutput"
 }
 
 function Get-RVersion {
     $rVersion = Run-Command "R --version | grep 'R version'" | Take-Part -Part 2
-    return "R $rVersion"
+    return "$rVersion"
 }
 
 function Get-RustVersion {
     $rustVersion = Run-Command "rustc --version" | Take-Part -Part 1
-    return "Rust $rustVersion"
+    return $rustVersion
 }
 
 function Get-RustfmtVersion {
     $version = Run-Command "rustfmt --version" | Take-Part -Part 1
-    return "Rustfmt $version"
+    return $version
 }
 
 function Get-RustdocVersion {
     $version = Run-Command "rustdoc --version" | Take-Part -Part 1
-    return "Rustdoc $version"
+    return $version
 }
 
 function Get-RustCargoVersion {
     $version = Run-Command "cargo --version" | Take-Part -Part 1
-    return "Cargo $version"
+    return $version
 }
 
 function Get-RustClippyVersion {
     $version = Run-Command "cargo clippy --version" | Take-Part -Part 1
-    return "Clippy $version"
+    return $version
 }
 
 function Get-Bindgen {
     $bindgenVersion = Run-Command "bindgen --version" | Take-Part -Part 1
-    return "Bindgen $bindgenVersion"
+    return $bindgenVersion
 }
 
 function Get-Cbindgen {
     $cbindgenVersion = Run-Command "cbindgen --version" | Take-Part -Part 1
-    return "Cbindgen $cbindgenVersion"
+    return $cbindgenVersion
 }
 
 function Get-Cargooutdated {
     $cargoOutdatedVersion = Run-Command "cargo outdated --version" | Take-Part -Part 1
-    return "Cargo-outdated $cargoOutdatedVersion"
+    return $cargoOutdatedVersion
 }
 
 function Get-Cargoaudit {
     $cargoAuditVersion = Run-Command "cargo-audit --version" | Take-Part -Part 1
-    return "Cargo-audit $cargoAuditVersion"
+    return $cargoAuditVersion
 }
 
 function Get-RustupVersion {
     $rustupVersion = Run-Command "rustup --version" | Select-Object -First 1 | Take-Part -Part 1
-    return "Rustup ${rustupVersion}"
+    return "${rustupVersion}"
 }
 
 function Get-VcpkgVersion {
     $vcpkgVersion = Run-Command "vcpkg version" | Select-Object -First 1 | Take-Part -Part 5 | Take-Part -Part 0 -Delimiter "-"
     $commitId = git -C "/usr/local/share/vcpkg" rev-parse --short HEAD
-    return "Vcpkg $vcpkgVersion (build from master \<$commitId>)"
+    return "$vcpkgVersion (build from master \<$commitId>)"
 }
 
-function Get-GccVersion {
+function Add-GccVersions {
+    param (
+        [HeaderNode] $HeaderNode
+    )
+
```

Comment:
Honestly, I would try to avoid passing nodes into the method.
What do you think about returning array of nodes from method?

```pwsh
function Get-GccVersions {
  $versionList = Get-ToolsetValue -KeyPath gcc.versions
  $versionList | Foreach-Object {
        $nameVersion = Run-Command "gcc-${_} --version" | Select-Object -First 1
        $version = $nameVersion.Substring("gcc-${_}".Length).Trim()
        return [ToolNode]::new("GCC ${_}", "$version - available by ``gcc-${_}`` alias")
  }
}
```

Then invoke function like
```pwsh
$node = [HeaderNode]::new(...)
Get-GccVersions | ForEach-Object { $node.AddNode($_) }
```

Or even implement new method for HeaderNode `AddNodes`

Reply:
I did as suggested. Also followed the second suggestion and implemented new method `AddNodes`